File: strategy/en_us/java/FahrenheitToCelsius.java
/**
 *	File: FahrenheitToCelsius
 *  Role: Concrete Strategy
 *  Purpose: Class responsible to convert a measure in Fahrenheit into Celsius.
 */
public class FahrenheitToCelsius implements Converter {

	/*
	 *	Convert a measure in Fahrenheit into Celsius.
	 *
	 *	@param currentMeasure	a measure in Fahrenheit.
	 *	@return	the measure converted into Celsius
	 */
	@Override
	public float convert(float currentMeasure) {

		/*
		 *	Converting Fahrenheit to Celsius.
		 *	For more information:
		 *		https://en.wikipedia.org/wiki/Conversion_of_units_of_temperature
		 */
		float convertedMeasure = (currentMeasure - 32f) * (5f / 9f);

		/*
		 *	Inform the conversion as follows: <X>°F (Fahrenheit) = <Y>°C (Celsius).
		 *	Used to show a result of an application using Strategy Pattern.
		 */
		System.out.println(currentMeasure + FAHRENHEIT + " = " + convertedMeasure + CELSIUS);

		return convertedMeasure;
	}
}

File: strategy/en_us/python/fahrenheit_to_celsius.py
"""Converter: A pseudo-Interface for the strategies"""
from converter import Converter

class FahrenheitToCelsius(Converter):
	"""
	Class responsible to convert a measure in Fahrenheit into Celsius.
	It is a Concrete Strategy of the context.
	"""

	FAHRENHEIT = "*F (Fahrenheit)"
	CELSIUS = "*C (Celsius)"

	def convert(self, current_measure):
		"""
		Convert a measure in Fahrenheit into Celsius.
		For more information:
			https://en.wikipedia.org/wiki/Conversion_of_units_of_temperature
		"""
		converted_measure = (current_measure - 32.0) * (5.0/9.0)

		print "%.3f %s = %.3f %s" % (current_measure, self.FAHRENHEIT,
			converted_measure, self.CELSIUS)

		return converted_measure
